### DISCUSSION TITLE
Doubt in Video 99, timestamp 15:31:20

### DISCUSSION THREAD

**Discussion by AJAY31797** [mrdbourke/pytorch-deep-learning#726] [Q&A]
In the developed model, the in_features of the first Linear layer is considered as input_shape. I am not able to understand why. The input will be first passed to the Flatten layer, which will eventually change its shape. As it is a sequential model, the output from Flatten layer should be input for the first Linear layer. So why are we considering the input_features of the first Linear layer as input_shape?

**Comment by shisirkha** [comment]
The `in_features` parameter of the first Linear layer in a PyTorch model is used to specify the number of input features that the layer expects. This value is typically set to the number of output features from the previous layer, which is equivalent to the size of the flattened tensor that is passed to the Linear layer.

In a sequential model, the output from the Flatten layer is indeed passed as input to the first Linear layer. However, the Flatten layer does not change the number of input features, but rather the shape of the input tensor. The Flatten layer takes a tensor of shape `(batch_size, channels, height, width)` and flattens it into a tensor of shape `(batch_size, channels * height * width)`. The number of input features remains the same, but the shape of the tensor changes.

Therefore, the `in_features` parameter of the first Linear layer should be set to the number of output features from the previous layer, which is equivalent to the number of input features to the Flatten layer. This ensures that the dimensions of the input tensor are compatible with the weight matrix of the Linear layer.

I hope this helps!
